core/commands/dag/dag.go

```diff
@@ -55,7 +55,8 @@ type ResolveOutput struct {
 
 // CarImportOutput is the output type of the 'dag import' commands
 type CarImportOutput struct {
-	Root RootMeta
+	BlockCount uint64
+	Root       *RootMeta
```

Comment:
I will defer to @aschmahmann but probably the easiest thing is to `*` them both and tag
```
`json:",omitempty"`
```

Reply:
cool, I didn't know if `*` was even an appropriate thing to do in this situation

Reply:
ok, done that, it's cleaned up the messages pretty nicely, but has introduced an exclusive union which I've added some checks for—mildly awkward but workable